crates/sui-core/src/consensus_adapter.rs

```diff
@@ -229,16 +231,25 @@ impl ConsensusAdapter {
         metrics: ConsensusAdapterMetrics,
     ) -> Arc<Self> {
         let num_inflight_transactions = Default::default();
+        let low_scoring_authorities = ArcSwap::new(Arc::new(HashMap::new()));
```

Comment:
I believe that this won't work as expected. The `ConsensusAdapter` practically is initialing its own version of the `ArcSwap` here . Same for the `ConsensusHandler` . So when we update the authorities in the `ConsensusHandler` via the `swap` method, all we do is to just update the values on that instance only.

Instead we should _share_ the `ArcSwap` across the components. To do that we should instead create an instance like `Arc<ArcSwap<HashMap...>>` , and practically share the `Arc` between them. Then once you perform a `swap` this will update across all the components.  Basically my original comment here https://github.com/MystenLabs/sui/pull/8847#discussion_r1125732446

@laura-makdah  could you please confirm that this could be the issue here?

Reply:
Great catch! That's explain the disconnect :)

Reply:
Thanks for the find! This was it. I had to do some refactoring to keep it as a HashMap instead of DashMap like I first had it.